### PR TITLE
ARROW-5434: [Memory][Java] Introduce wrappers for backward compatibility.

### PR DESCRIPTION
**Pull request by praveenbingo**
- Introduce some wrapper methods to reduce amount of client changes.
- Changes were introduced as part of patch to support arrow buffers on random memory.

### CODE REVIEW

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java [hunk 1]
```diff
@@ -1204,4 +1204,45 @@ public ArrowBuf reallocIfNeeded(final int size) {
               "Realloc is only available in the context of operator's UDFs");
     }
   }
+
+  /**
+   * Following are wrapper methods to keep this backward compatible.
+   */
+  public void release() {
```
Comment (emkornfield) [inline java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: can we mark these as "@deprecated", so clients know they should moving off of them?

**Comment by emkornfield** [comment]
@praveenbingo I think we prefer creating new issues instead of multiple PRs attached to the same issue.  Do you mind creating a new one?

**Comment by praveenbingo** [comment]
> @praveenbingo I think we prefer creating new issues instead of multiple PRs attached to the same issue. Do you mind creating a new one?

Sure will do.

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java (hunk 1 above)
Comment (praveenbingo) [reply java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: done.

Comment (pravindra) [reply java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: why deprecate ? It's a useful convenience fn.

File: java/memory/src/main/java/io/netty/buffer/ArrowBuf.java [hunk 2]
```diff
@@ -1204,4 +1204,51 @@ public ArrowBuf reallocIfNeeded(final int size) {
...
+  }
+
+  @Deprecated
+  public void retain(int increment) {
+    referenceManager.retain(increment);
+  }
+
+  @Deprecated
+  public ArrowBuf clear() {
+    this.readerIndex = this.writerIndex = 0;
```
Comment (pravindra) [inline java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: agree this should be deprecated - readerIndex and writerIndex should move to NettyArrowBuf

Comment (praveenbingo) [reply java/memory/src/main/java/io/netty/buffer/ArrowBuf.java]: ideally we should not have duplicate methods doing the same work, at some point in the future we should remove the older..

**Comment by praveenbingo** [comment]
@pravindra can you please merge if this look to you.

**Review by pravindra — APPROVED** [review]

**Pull request closed without merging**